src/coffee/export.coffee

```diff
@@ -112,14 +113,21 @@ class Export
   # return the correct product service in case query string is used or not
   _getProductService: (staged = true, customWherePredicate = false) ->
     productsService = @client.productProjections
-    if customWherePredicate
-      productsService.where(customWherePredicate)
+    perPage = 100
 
     if @queryOptions.queryString
-      productsService.byQueryString(@queryOptions.queryString, @queryOptions.isQueryEncoded)
+      query = @queryOptions.queryString
+      if @queryOptions.isQueryEncoded
+        query = decodeURIComponent(query)
+
+      if customWherePredicate
+        query = @_appendPredicateToQueryString(query, customWherePredicate)
+
+      productsService.byQueryString(query, false)
       productsService
     else
-      productsService.all().perPage(500).staged(staged)
+      productsService.where(customWherePredicate || '')
+      productsService.all().perPage(perPage).staged(staged)
```

Comment:
Can you explain exactly why this didn't work and how this should solve the problem? Is it a problem here or in the sdk?